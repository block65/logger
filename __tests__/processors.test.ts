import {
  afterEach,
  beforeEach,
  describe,
  expect,
  jest,
  test,
} from '@jest/globals';
import { createLoggerWithWaitableMock } from './helpers.js';

describe('Processsors', () => {
  beforeEach(() => {
    jest.useFakeTimers({ now: new Date('2009-02-13T23:31:30.000Z') });
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  test('Crashy', async () => {
    const [logger, callback, errback] = createLoggerWithWaitableMock({
      processors: [
        () => {
          throw new Error('oops I crashed');
        },
      ],
    });

    logger.error('I sure hope this doesnt crash');

    await expect(callback.waitUntilCalled()).resolves.toMatchSnapshot();

    expect(errback).toBeCalled();
    expect(errback.mock.calls).toMatchSnapshot();
  });
});
