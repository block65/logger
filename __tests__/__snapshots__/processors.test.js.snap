// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processsors Crashy 1`] = `
Array [
  Object {
    "level": 50,
    "msg": "I sure hope this doesnt crash",
    "time": 2009-02-13T23:31:30.000Z,
  },
]
`;

exports[`Processsors Crashy 2`] = `
Array [
  Array [
    [Error: oops I crashed],
  ],
]
`;
